Flora/Flora/View Controllers/SearchViewController.m

```diff
@@ -116,11 +78,56 @@ - (IBAction)onTapSearch:(id)sender {
     self.searchButton.enabled = NO;
     [self queryPlants];
 }
+- (IBAction)tappedUseLocation:(id)sender {
+    // pull API Key from your new Keys.plist file
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
 
-- (IBAction)onTapLocation:(id)sender {
-    [self presentViewController:self.locationAlert animated:YES completion:nil];
+    NSString *key = [dict objectForKey: @"googlePlace"];
+    [GMSPlacesClient provideAPIKey:key];
+    
+    GMSAutocompleteViewController *acController = [[GMSAutocompleteViewController alloc] init];
+    acController.delegate = self;
+    
+    // Specify the place data types to return.
+    GMSPlaceField fields = (GMSPlaceFieldCoordinate | GMSPlaceFieldFormattedAddress);
+    acController.placeFields = fields;
+    
+    _filter = [[GMSAutocompleteFilter alloc] init];
+    _filter.type = kGMSPlacesAutocompleteTypeFilterAddress;
+    acController.autocompleteFilter = _filter;
+    
+    // Display the autocomplete view controller.
+    [self presentViewController:acController animated:YES completion:nil];
 }
-- (IBAction)tappedUseLocation:(id)sender {
+
+- (IBAction)locationSegmentControl:(id)sender {
+    if(self.locationControl.selectedSegmentIndex == 1) {
+        if(_lat) {
+            [self.moistureControl setUserInteractionEnabled:NO];
+            [self.temperaturecontrol setUserInteractionEnabled:NO];
+            [self.sunlightControl setUserInteractionEnabled:NO];
+            [[APIManager shared] weatherValuesAtLat:self.lat atLong:self.lon withCompletion:^(int moist, int sun, int temp, NSError * _Nonnull error) {
+                if(error) {
+                    NSLog(@"Error getting weather values: %@", error.localizedDescription);
```

Comment:
let's post some message to the user if their request has failed...  and since was 'triggered' by the selection.. maybe just set the self.locationControl.selectedSegmentIndex back to zero.

So they can keep 'tapping' the segment to retry the api call?